controllers/client/round_tripper.go

```diff
@@ -48,3 +55,14 @@ func (in *instrumentedRoundTripper) RoundTrip(r *http.Request) (*http.Response,
 	}
 	return resp, err
 }
+
+func (in *instrumentedRoundTripper) addHeaders(headers map[string]string) {
+	if in.headers == nil {
+		in.headers = headers
+		return
+	}
+
+	for k, v := range headers {
+		in.headers[k] = v
+	}
+}
```

Comment:
```suggestion
	if headers == nil {
		return
	}

	if in.headers == nil {
		in.headers = map[string]string{}
	}

	for k, v := range headers {
 		in.headers[k] = v
 	}
```

I think it would be useful to add an extra check on `headers` to make sure we don't take any action if it's `nil`.
I've added an example how I would potentially do it to make sure both src and dst are not-nil.

Reply:
I took your example with a minor difference on map initialization to make it visually distinct from declaration